kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -2932,9 +2938,22 @@ bool IndexerASTVisitor::VisitFunctionDecl(clang::FunctionDecl* Decl) {
     }
   }
 
-  MaybeRecordDefinitionRange(NameRangeInContext, OuterNode,
-                             BuildNodeIdForDefnOfDecl(Decl));
   bool IsFunctionDefinition = IsDefinition(Decl);
+  if (!IsFunctionDefinition || Decl->isImplicit() ||
+      Decl->getBody() == nullptr) {
+    MaybeRecordDefinitionRange(NameRangeInContext, OuterNode,
+                               BuildNodeIdForDefnOfDecl(Decl));
```

Comment:
I think this would be simpler as two separate calls, where the full definition is only emitted when found, rather than repeating the same function twice with an optional parameter that isn't used anywhere else.

Reply:
Done